### DISCUSSION TITLE
How to preserve revision tags during istioctl install?

### DISCUSSION THREAD

**Discussion by shokada** [istio/istio#48779] [Q&A]
**Question**
I've encountered a situation where I need to update Istio configuration using `istioctl install` without upgrading the Istio version itself. However, each time I run `istioctl install`, the revision tags set by `istioctl tag set` seem to be lost. This presents a problem if pods are started during the installation process, as they might not receive the correct sidecar injection without the revision tags.

Is there a recommended approach to update Istio configuration using `istioctl install` while ensuring that revision tags are preserved? Or is there a way to prevent potential issues with sidecar injection during the update process?

**Attempted Solutions**
- I've considered performing a canary upgrade, but it seems like an overkill for minor configuration changes and not a version upgrade.
- I've tried looking for a flag or command option that could help with this issue but haven't found one.

**Comment by hanxiaop** [comment]
What is your istioctl version?

**Comment by shokada** [comment]
@hanxiaop Thank you for your response. Here are the version details for my Istio setup:
```
-> % istioctl version -i infra-istio-system
client version: 1.20.1
control plane version: 1.20.1
data plane version: 1.20.1 (74 proxies)
```

**Comment by hanxiaop** [comment]
@shokada Just found a bug, the fix should be available in the next patch release, 1.20.3. Before the patch, I think you can delete the operator-related labels that contain `operator.istio.io/` or `install.operator.istio.io/` from the tag webhook you've created (should be named `istio-revision-tag-*`) to preserve the webhook after re-installation as a workaround.

**Comment by shokada** [comment]
@hanxiaop Understood regarding the bug, and I appreciate the update. I'll wait for the patch release 1.20.3 to be available.
Thank you also for the suggested workaround. I will consider applying it if necessary.
